docs/modules/ROOT/pages/api/account.adoc

```diff
@@ -0,0 +1,158 @@
+[[ISRC6-__validate__]]
+==== `[.contract-item-name]#++__validate__++#++(calls: Array<Call>) → felt252++` [.item-kind]#external#
+
+Validates a transaction before execution.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[ISRC6-is_valid_signature]]
+==== `[.contract-item-name]#++is_valid_signature++#++(hash: felt252, signature: Array<felt252>) → felt252++` [.item-kind]#external#
+
+Validates whether a signature is valid or not for the given message hash.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract]
+[[Account]]
+=== `++Account++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/account/account.cairo#L27[{github-icon},role=heading-link]
+
+[.hljs-theme-light.nopadding]
+```javascript
+use openzeppelin::account::Account;
+```
+
+Base Account contract implementation.
+
+
+[.contract-index]
+.Functions
+--
+* xref:#Account-\\__validate_declare__[`++__validate_declare__(class_hash)++`]
+* xref:#Account-\\__validate_deploy__[`++__validate_deploy__(hash, signature)++`]
+* xref:#Account-set_public_key[`++set_public_key(new_public_key)++`]
+* xref:#Account-get_public_key[`++get_public_key()++`]
+
+[.contract-subindex-inherited]
+.ISRC6
+
+* xref:#Account-\\__execute__[`++__execute__(calls)++`]
+* xref:#Account-\\__validate__[`++__validate__(calls)++`]
+* xref:#Account-is_valid_signature[`++is_valid_signature(hash, signature)++`]
+
+[.contract-subindex-inherited]
+.ISRC5
+
+* xref:#Account-supports_interface[`++supports_interface(interface_id)++`]
+--
+
+[.contract-index]
+.Events
+--
+* xref:#Account-OwnerAdded[`++OwnerAdded(new_owner_guid)++`]
+* xref:#Account-OwnerRemoved[`++OwnerRemoved(removed_owner_guid)++`]
+--
+
+[.contract-item]
+[[Account-__execute__]]
+==== `[.contract-item-name]#++__execute__++#++(calls: Array<Call>) → Array<Span<felt252>>++` [.item-kind]#external#
+
+See xref:ISRC6-\\__execute__[ISRC6::\\__execute__].
+
+[.contract-item]
+[[Account-__validate__]]
+==== `[.contract-item-name]#++__validate__++#++(calls: Array<Call>) → felt252++` [.item-kind]#external#
+
+See xref:ISRC6-\\__validate__[ISRC6::\\__validate__].
+
+[.contract-item]
+[[Account-__validate_declare__]]
+==== `[.contract-item-name]#++__validate_declare__++#++(class_hash: felt252) → felt252++` [.item-kind]#external#
+
+Validates a Declare transaction.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[Account-__validate_deploy__]]
+==== `[.contract-item-name]#++__validate_deploy__++#++(class_hash: felt252, contract_address_salt: felt252, _public_key: felt252) → felt252++` [.item-kind]#external#
+
+Validates a DeployAccount transaction.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[Account-set_public_key]]
+==== `[.contract-item-name]#++set_public_key++#++(new_public_key: felt252)++` [.item-kind]#external#
+
+Sets a new public key for the account.
```

Comment:
I think we should mention that this emits `OwnerAdded` and `OwnerRemoved` and also that this can only be called by self

Reply:
Indeed, will udpdate.